python/lsst/daf/butler/registry/interfaces/_dimensions.py

```diff
@@ -48,6 +57,74 @@ class DimensionRecordStorage(ABC):
     subclasses to provide them is not a significant burden.
     """
 
+    @classmethod
+    @abstractmethod
+    def initialize(cls, db: Database, element: DimensionElement, *,
+                   context: Optional[StaticTablesContext] = None) -> DimensionRecordStorage:
+        """Construct an instance of this class using a standardized interface.
```

Comment:
This is an odd docstring, because the words you used also exactly describe new/init

Reply:
Not quite - `__new__` (which is what this is more like) is not usually expected to have the same signature (what I mean by "standardized") across all subclasses of an ABC, and that's also what makes `__new__` inappropriate (and maybe disallowed?) as an `abstractmethod`.